lib/ansible/modules/cloud/amazon/elasticsearch.py

```diff
@@ -0,0 +1,421 @@
+    if compare_policies(existing_access_json, supplied_access_json):
+        modifications_needed['AccessPolicies'] = module.params['access_policies']
+
+    return modifications_needed
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            domain={'required': True},
+            state={'required': True, 'choices': ['present', 'absent']},
+            version={'default': '5.1'},
+            instance_count={'type': 'int', 'default': 1},
+            instance_type={'required': True},
+            zone_awareness={'type': 'bool', 'default': True},
+            dedicated_master_enabled={
+                'type': 'bool', 'default': False
+            },
+            dedicated_master_type={},
+            dedicated_master_count={'type': 'int'},
+            ebs_volume_type={
+                'default': 'gp2', 'choices': ['standard', 'gp2', 'io1']
+            },
+            ebs_volume_size={'required': True, 'type': 'int'},
+            access_policies={'required': True},
+            wait={'type': 'bool'},
+            tags={'type': 'dict'},
+        )
+    )
+
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+    )
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(
+        module, boto3=True
+    )
+
+    client = boto3_conn(
+        module, conn_type='client', resource='es', region=region,
+        endpoint=ec2_url, **aws_connect_kwargs
+    )
+
+    if module.params['state'] not in ['present', 'absent']:
+        module.fail_json(msg='"state" must be "present" or "absent"')
+
+    if module.params['state'] == 'absent':
+        changed = ensure_deleted(client, module)
+        module.exit_json(changed=changed)
+
+    elif module.params['state'] == 'present':
+        changed, domain = ensure_present(client, module)
+        modifications_needed = check_existing_config(module, domain)
+
+        if modifications_needed:
+            changed = True
+            modifications_needed['DomainName'] = module.params['domain']
+            client.update_elasticsearch_domain_config(**modifications_needed)
+
+            # Reload after modifying
+            domain = is_present(client, module)
+
+        if module.params['tags']:
+            arn = domain['DomainStatus']['ARN']
+
+            tag_list = []
+            for each_key, each_value in module.params['tags'].items():
+                tag_list.append({'Key': each_key, 'Value': each_value})
+
+            client.add_tags(ARN=arn, TagList=tag_list)
+
+        if module.params['wait']:
+            while not domain['DomainStatus'].get('Endpoint'):
+                time.sleep(10)
+                domain = is_present(client, module)
+
+        domain_status = domain['DomainStatus']
+        cluster_config_status = domain_status['ElasticsearchClusterConfig']
+        ebs_options = domain_status['EBSOptions']
+
+        module.exit_json(
+            changed=changed,
+            arn=domain_status['ARN'],
```

Comment:
It's more common to use the helper camel_dict_to_snake_dict here https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#camel_dict_to_snake_dict.

You can just do `camel_dict_to_snake_dict(**domain_status)`